File: app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingInteractorImpl.java
package com.esoxjem.movieguide.listing;

import com.esoxjem.movieguide.Movie;
import com.esoxjem.movieguide.MoviesWraper;
import com.esoxjem.movieguide.network.TmdbWebService;

import java.util.List;

import io.reactivex.Observable;

/**
 * @author arun
 */
public class MoviesListingInteractorImpl implements MoviesListingInteractor {

    private TmdbWebService tmdbWebService;

    public MoviesListingInteractorImpl(TmdbWebService tmdbWebService) {
        this.tmdbWebService = tmdbWebService;
    }

    @Override
    public Observable<List<Movie>> fetchMovies() {
        return tmdbWebService.highestRatedMovies().map(MoviesWraper::getMovieList);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingInteractorImpl.kt
package com.esoxjem.movieguide.listing

import com.esoxjem.movieguide.Movie
import com.esoxjem.movieguide.MoviesWraper
import com.esoxjem.movieguide.network.TmdbWebService

import io.reactivex.Observable

/**
 * @author arun
 */
class MoviesListingInteractorImpl(private val tmdbWebService: TmdbWebService) : MoviesListingInteractor {

    override fun fetchMovies(): Observable<List<Movie>> {
        return tmdbWebService.highestRatedMovies().map { it.movieList }
    }
}
